llvm/lib/IR/Instruction.cpp

```diff
@@ -461,6 +463,29 @@ bool Instruction::hasPoisonGeneratingMetadata() const {
          hasMetadata(LLVMContext::MD_align);
 }
 
+bool Instruction::hasLoopMetadataOtherThanDebugLoc() const {
```

Comment:
We usually use the term "NonDebug" for this sort of situation (see: https://github.com/llvm/llvm-project/blob/8ec819e62c22c6086837eb653ae91e76b0e0842b/llvm/include/llvm/IR/Instruction.h#L860, https://github.com/llvm/llvm-project/blob/8ec819e62c22c6086837eb653ae91e76b0e0842b/llvm/include/llvm/IR/Instruction.h#L445, etc)

So probably "hasNonDebugLoopMetadata" might be a good name for this function?

Though it doesn't look like we have a generic way to query "is this MDOperand a debug operand", so just testing for DILocation specifically probably suffices.

Reply:
Thanks for the review! I've renamed the method to `hasNonDebugLocMetadata`, to conform more closely to the standard, yet still express that we're specifically looking at debug locations. Does that work?